README.md

```diff
@@ -1 +1,42 @@
-# datalog-parser
+# datalog-parser <a href="https://clojars.org/io.lambdaforge/datalog-parser"> <img src="https://img.shields.io/clojars/v/io.lambdaforge/datalog-parser.svg" /></a> 
+
+Datahike's form parser.
+
+## Usage
+Add `[io.lambdaforge/datalog-parser "0.0.1"]` to your `project.clj`. Start a repl and run:
+
+```Clojure
+(require '[datalog-parser.core :as dp])
+
+(def query '[:find ?x :in $ ?y :where [?x :z ?y]])
+
+(dp/parse-query query)
+
+;;=>
+;;#datalog-parser.coreQuery{:qfind #datalog-parser.FindRel{:elements [#datalog-parser.Variable{:symbol ?x}]},
```

Comment:
I can imagine that this protocol based `-postwalk` and `deftrecord` were introduced because of performance reasons from the start. In Clojure(Script) code the datalog query has to be parsed everytime a database is accessed. In websites embedding a datascript parser performance could be considerable compared to actually accessing a simple index. Overall @tonsky has put considerable effort into making datascript fast. I have found no direct proof for this claim in the git history of datascript's codebase though.

Reply:
@whilo It's clear a lot of the choices were guided by performance concerns.  I tried to leave these unmolested, except when the benefit seemed marginal in comparison to the idiosyncrasy (e.g. converting uses of `vec` into ` into []`, and compacting the implementation to `reverse-ref`)

Reply:
Obviously protocol dispatch is much faster than multimethods - at least on the JVM - but given there's a closed set of elements, i would personally have probably implemented an internal macro which looks like `defmethod` but which builds up a constant-time dispatch fn by adding clauses to a `case`, or something like that, and kept everything pure-data.  Not super opposed to the impl., just not how I would have done it, and a high cost to pay for performance which could be attained by other means.

Reply:
Right, I think it could be definitely improved in this direction. It is odd that postwalk is patched like that and it is unclear what the performance trade-offs are really. I was just speculating.